freqtrade/finance/wallets.py

```diff
@@ -0,0 +1,41 @@
+# pragma pylint: disable=W0603
+""" Wallet """
+import logging
+from typing import Dict
+from collections import namedtuple
+from freqtrade.exchange import Exchange
+
+logger = logging.getLogger(__name__)
+
+
+class Wallets(object):
+
+    # wallet data structure
+    wallet = namedtuple(
+        'wallet',
+        ['exchange', 'currency', 'free', 'used', 'total']
+    )
+
+    def __init__(self, exchange: Exchange) -> None:
+        self.exchange = exchange
+        self.wallets: Dict[str, self.wallet] = {}
+        self._update_wallets()
+
+    def _update_wallets(self) -> None:
+        balances = self.exchange.get_balances()
+
+        for currency in balances:
+            info = {
+                'exchange': self.exchange.id,
+                'currency': currency,
+                'free': balances[currency]['free'],
+                'used': balances[currency]['used'],
+                'total': balances[currency]['total']
```

Comment:
i am not 100% certain, but i think this is playing with fire - i'm not sure if all 3 elements are available for all exchanges all the time ... 
i would add an exception handler for "indexError" to avoid having this crash the bot ... ?

Reply:
From CCXT: "Some exchanges do not return the full set of balance information from their API. Those will only return just the free or just the total funds, i.e. funds used on orders unknown. In such cases ccxt will try to obtain the missing data from .orders cache and will guess complete balance info from what is known for sure. However, in rare cases the available info may not be enough to deduce the missing part, thus, the user shoud be aware of the possibility of not getting complete balance info from less sophisticated exchanges."

I need only 'free' for opening a trade. So your idea is to get 'free' and catch indexError of others? In case even 'free' not available then we agree that the bot should crash?

Reply:
this seems the only thing not fixed so far ... 
i think it's dangerous - ccxt seems to unify balance "more or less" - but on different exchanges contains different additional attributes, so the loop may loop over something "not currency" ... which may not be a problem but could crash the bot (we've also seen that with ICO coins a while ago ...)

sample: binance `fetch_currency()` returns many more keys than the same call against kraken does ...

Reply:
Yes, it is not fixed yet ! give me second !!! :D